app/javascripts/main/fileUtils.js

```diff
@@ -1,117 +1,124 @@
-var { app } = require('electron');
-var fs = require('fs');
-var path = require('path');
-var request = require("request");
-var appPath = app.getPath('userData');
+import yauzl from "yauzl";
+const fs = require("fs");
+const path = require("path");
 
-export class FileUtils {
-  readJSONFile(path, callback) {
-    fs.readFile(path, 'utf8', function (err, data) {
-      if (err) {
-        console.error("Unable to read JSON file", path);
-        callback(null, err);
-        return;
-      }
-      var obj = JSON.parse(data);
-      callback(obj);
-    });
-  }
-
-  writeJSONFile(data, path, callback) {
-    this.ensureDirectoryExists(path);
-    fs.writeFile(path, JSON.stringify(data, null, 2), 'utf8', (err) => {
-      callback(err);
-    });
-  }
+export const FILE_DOES_NOT_EXIST = "ENOENT";
+export const FILE_ALREADY_EXISTS = "EEXIST";
 
-  deleteAppRelativeDirectory(relativePath) {
-    console.log("Delete App Relative Directory", relativePath);
-    const deleteDirectory = (dirPath) => {
-      if (fs.existsSync(dirPath)) {
-        fs.readdirSync(dirPath).forEach((entry) => {
-          const entryPath = path.join(dirPath, entry);
-          if (fs.lstatSync(entryPath).isDirectory()) {
-            deleteDirectory(entryPath);
-          } else {
-            fs.unlinkSync(entryPath);
-          }
-        });
-        fs.rmdirSync(dirPath);
-      }
-    };
+export async function readJSONFile(path) {
+  const data = await fs.promises.readFile(path, "utf8");
+  return JSON.parse(data);
+}
 
-    deleteDirectory(path.join(appPath, relativePath));
-  }
+export async function writeJSONFile(filepath, data) {
+  await ensureDirectoryExists(path.dirname(filepath));
+  await fs.promises.writeFile(filepath, JSON.stringify(data, null, 2), "utf8");
+}
 
-  ensureDirectoryExists(filePath) {
-    var dirname = path.dirname(filePath);
-    if (fs.existsSync(dirname)) {
-      return true;
+export async function ensureDirectoryExists(dirPath) {
+  try {
+    const stat = await fs.promises.lstat(dirPath);
+    if (!stat.isDirectory()) {
+      throw new Error(
+        "Tried to create a directory where a file of the same " +
+          `name already exists: ${dirPath}`
+      );
+    }
+  } catch (error) {
+    if (error.code === FILE_DOES_NOT_EXIST) {
+      // No directory here. Make sure there is a *parent* directory, and then
```

Comment:
I know it's a matter of preference, but we've been using /** */ style for inline comments. It forces us to be more "design"-aware with our comments which can make them more thoughtful. And it's more aesthetic.